Ban event doesn't work

### Skript/Server Version

```markdown
[Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[Skript] Server Version: 1.21.11-132-c5eb079 (MC: 1.21.11)
[Skript] Skript Version: 2.15.3 (skriptlang-github)
[Skript] Installed Skript Addons: None
[Skript] Installed dependencies: 
[Skript]  - Vault v1.7.3-b131
[Skript]  - WorldGuard v7.0.17+2370-e42d8bc
```

### Bug Description

I have recently made a /ban command for my server, but whenever i try testing it, it says that an unexpected error has occurred, followed by a large error message in my log.

### Expected Behavior

It is supposed to ban the player/kick them from the server.

### Steps to Reproduce

Try making a ban command. Then try executing the command. It will say that an unexpected problem has occurred, followed by the error log.

### Errors or Screenshots

Error log:
```
[18:40:32] [Server thread/ERROR]: #!#!
--
174 | [18:40:32] [Server thread/ERROR]: #!#! [Skript] Severe Error:
175 | [18:40:32] [Server thread/ERROR]: #!#!
176 | [18:40:32] [Server thread/ERROR]: #!#! An unexpected error occurred with Skript. This issue is likely not your fault.
177 | [18:40:32] [Server thread/ERROR]: #!#! Report the issue: https://github.com/SkriptLang/Skript/issues
178 | [18:40:32] [Server thread/ERROR]: #!#!
179 | [18:40:32] [Server thread/ERROR]: #!#! Stack trace:
180 | [18:40:32] [Server thread/ERROR]: #!#! Caused by: java.lang.NullPointerException: input
181 | [18:40:32] [Server thread/ERROR]: #!#!     at java.base/java.util.Objects.requireNonNull(Objects.java:246)
182 | [18:40:32] [Server thread/ERROR]: #!#!     at net.kyori.adventure.text.flattener.ComponentFlattenerImpl.flatten0(ComponentFlattenerImpl.java:117)
183 | [18:40:32] [Server thread/ERROR]: #!#!     at net.kyori.adventure.text.flattener.ComponentFlattenerImpl.flatten(ComponentFlattenerImpl.java:113)
184 | [18:40:32] [Server thread/ERROR]: #!#!     at net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializerImpl.serialize(LegacyComponentSerializerImpl.java:273)
185 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//org.skriptlang.skript.bukkit.text.TextComponentParser.toLegacyString(TextComponentParser.java:580)
186 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//org.skriptlang.skript.bukkit.entity.player.elements.effects.EffBan.execute(EffBan.java:102)
187 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.Effect.run(Effect.java:42)
188 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:40)
189 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:67)
190 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:33)
191 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.execute2(ScriptCommand.java:353)
192 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.lambda$execute$0(ScriptCommand.java:312)
193 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.execute(ScriptCommand.java:323)
194 | [18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.onCommand(ScriptCommand.java:258)
195 | [18:40:32] [Server thread/ERROR]: #!#!     at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
196 | [18:40:32] [Server thread/ERROR]: #!#!     at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigCommand.run(BukkitCommandNode.java:83)
197 | [18:40:32] [Server thread/ERROR]: #!#!     at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73)
198 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:30)
199 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:13)
200 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8)
201 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.CommandQueueEntry.execute(CommandQueueEntry.java:5)
202 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:104)
203 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:469)
204 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.Commands.performCommand(Commands.java:374)
205 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.Commands.performCommand(Commands.java:362)
206 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.performUnsignedChatCommand(ServerGamePacketListenerImpl.java:2375)
207 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$12(ServerGamePacketListenerImpl.java:2348)
208 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.TickTask.run(TickTask.java:18)
209 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:177)
210 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24)
211 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1525)
212 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:188)
213 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:151)
214 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1505)
215 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1499)
216 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.recordTaskExecutionTimeWhileWaiting(MinecraftServer.java:1225)
217 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1341)
218 | [18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:388)
219 | [18:40:32] [Server thread/ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1474)
220 | [18:40:32] [Server thread/ERROR]: #!#!
221 | [18:40:32] [Server thread/ERROR]: #!#! Skript: 2.15.3 (latest)
222 | [18:40:32] [Server thread/ERROR]: #!#!     Flavor: skriptlang-github
223 | [18:40:32] [Server thread/ERROR]: #!#!     Date: 15:56:00.516894
224 | [18:40:32] [Server thread/ERROR]: #!#! Bukkit: 1.21.11-R0.1-SNAPSHOT
225 | [18:40:32] [Server thread/ERROR]: #!#! Minecraft: 1.21.11
226 | [18:40:32] [Server thread/ERROR]: #!#! Java: 25.0.3 (OpenJDK 64-Bit Server VM 25.0.3+9-LTS)
227 | [18:40:32] [Server thread/ERROR]: #!#! OS: Linux amd64 6.8.0-124-generic
228 | [18:40:32] [Server thread/ERROR]: #!#!
229 | [18:40:32] [Server thread/ERROR]: #!#! Server platform: Paper
230 | [18:40:32] [Server thread/ERROR]: #!#!
231 | [18:40:32] [Server thread/ERROR]: #!#! Current node: null
232 | [18:40:32] [Server thread/ERROR]: #!#! Current item: ban the 1st argument
233 | [18:40:32] [Server thread/ERROR]: #!#! Current trigger: command /ban (simple event) (staff.sk, line 171)
234 | [18:40:32] [Server thread/ERROR]: #!#! Thread: Server thread
235 | [18:40:32] [Server thread/ERROR]: #!#! Language: english
236 | [18:40:32] [Server thread/ERROR]: #!#! Link parse mode: DISABLED
237 | [18:40:32] [Server thread/ERROR]: #!#! End of Error.
238 | [18:40:32] [Server thread/ERROR]: #!#!

[18:40:32] [Server thread/ERROR]: #!#! 
[18:40:32] [Server thread/ERROR]: #!#! [Skript] Severe Error:
[18:40:32] [Server thread/ERROR]: #!#! 
[18:40:32] [Server thread/ERROR]: #!#! An unexpected error occurred with Skript. This issue is likely not your fault.
[18:40:32] [Server thread/ERROR]: #!#! Report the issue: https://github.com/SkriptLang/Skript/issues
[18:40:32] [Server thread/ERROR]: #!#! 
[18:40:32] [Server thread/ERROR]: #!#! Stack trace:
[18:40:32] [Server thread/ERROR]: #!#! Caused by: java.lang.NullPointerException: input
[18:40:32] [Server thread/ERROR]: #!#!     at java.base/java.util.Objects.requireNonNull(Objects.java:246)
[18:40:32] [Server thread/ERROR]: #!#!     at net.kyori.adventure.text.flattener.ComponentFlattenerImpl.flatten0(ComponentFlattenerImpl.java:117)
[18:40:32] [Server thread/ERROR]: #!#!     at net.kyori.adventure.text.flattener.ComponentFlattenerImpl.flatten(ComponentFlattenerImpl.java:113)
[18:40:32] [Server thread/ERROR]: #!#!     at net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializerImpl.serialize(LegacyComponentSerializerImpl.java:273)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//org.skriptlang.skript.bukkit.text.TextComponentParser.toLegacyString(TextComponentParser.java:580)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//org.skriptlang.skript.bukkit.entity.player.elements.effects.EffBan.execute(EffBan.java:102)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.Effect.run(Effect.java:42)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:40)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:67)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:33)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.execute2(ScriptCommand.java:353)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.lambda$execute$0(ScriptCommand.java:312)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.execute(ScriptCommand.java:323)
[18:40:32] [Server thread/ERROR]: #!#!     at Skript-2.15.3.jar//ch.njol.skript.command.ScriptCommand.onCommand(ScriptCommand.java:258)
[18:40:32] [Server thread/ERROR]: #!#!     at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[18:40:32] [Server thread/ERROR]: #!#!     at io.papermc.paper.command.brigadier.bukkit.BukkitCommandNode$BukkitBrigCommand.run(BukkitCommandNode.java:83)
[18:40:32] [Server thread/ERROR]: #!#!     at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:30)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:13)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.UnboundEntryAction.lambda$bind$0(UnboundEntryAction.java:8)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.CommandQueueEntry.execute(CommandQueueEntry.java:5)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.execution.ExecutionContext.runCommandQueue(ExecutionContext.java:104)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.Commands.executeCommandInContext(Commands.java:469)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.Commands.performCommand(Commands.java:374)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.commands.Commands.performCommand(Commands.java:362)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.performUnsignedChatCommand(ServerGamePacketListenerImpl.java:2375)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChatCommand$12(ServerGamePacketListenerImpl.java:2348)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.TickTask.run(TickTask.java:18)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:[177](https://aternos.org/log/#L177))
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1525)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:188)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:151)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1505)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1499)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.recordTaskExecutionTimeWhileWaiting(MinecraftServer.java:1225)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1341)
[18:40:32] [Server thread/ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:388)
[18:40:32] [Server thread/ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1474)
[18:40:32] [Server thread/ERROR]: #!#! 
[18:40:32] [Server thread/ERROR]: #!#! Skript: 2.15.3 (latest)
[18:40:32] [Server thread/ERROR]: #!#!     Flavor: skriptlang-github
[18:40:32] [Server thread/ERROR]: #!#!     Date: 15:56:00.516894
[18:40:32] [Server thread/ERROR]: #!#! Bukkit: 1.21.11-R0.1-SNAPSHOT
[18:40:32] [Server thread/ERROR]: #!#! Minecraft: 1.21.11
[18:40:32] [Server thread/ERROR]: #!#! Java: 25.0.3 (OpenJDK 64-Bit Server VM 25.0.3+9-LTS)
[18:40:32] [Server thread/ERROR]: #!#! OS: Linux amd64 6.8.0-124-generic
[18:40:32] [Server thread/ERROR]: #!#! 
[18:40:32] [Server thread/ERROR]: #!#! Server platform: Paper
[18:40:32] [Server thread/ERROR]: #!#! 
[18:40:32] [Server thread/ERROR]: #!#! Current node: null
[18:40:32] [Server thread/ERROR]: #!#! Current item: ban the 1st argument
[18:40:32] [Server thread/ERROR]: #!#! Current trigger: command /ban (simple event) (staff.sk, line 171)
[18:40:32] [Server thread/ERROR]: #!#! Thread: Server thread
[18:40:32] [Server thread/ERROR]: #!#! Language: english
[18:40:32] [Server thread/ERROR]: #!#! Link parse mode: DISABLED
[18:40:32] [Server thread/ERROR]: #!#! End of Error.
[18:40:32] [Server thread/ERROR]: #!#!
```

### Other

Also, unbanning players (atleast on my command, which broadcasts as to who has been unbanned), it says "<lang:container.crafting>" instead of an offline player.

Here is the exact skript I use as well to ban players:
```
command /ban [<offline player>] [<text>]:
	permission: larpchy.staff.ban
	trigger:
		if arg-1 is not set:
			send "&c[*] You have to input a username to ban."
		else:
			if arg-2 is not set:
				ban arg-1
				kick arg-1 with reason "&c&l[LarpchyMC] &cYou have been banned."
				broadcast "&a[*] %arg-1's display name% has been banned by %player's display name%."
			else:
				ban arg-1
				kick arg-1 with reason "&c&l[LarpchyMC] &cYou have been banned. Reason: '%arg-2%.'"
				broadcast "&a[*] %arg-1's display name% has been banned by %player's display name%. Reason: %arg-2%"

command /unban [<offline player>] [<text>]:
	permission: larpchy.staff.unban
	trigger:
		if arg-1 is not set:
			send "&c[*] You have to input a username to unban."
		else:
			if arg-2 is not set:
				unban arg-1
				broadcast "&a[*] %arg-1's display name% has been unbanned by %player's display name%."
			else:
				unban arg-1
				broadcast "&a[*] %arg-1's display name% has been unbanned by %player's display name%. Reason: %arg-2%"
```

### Agreement

- [x] I have read the guidelines above and affirm I am following them with this report.

--- a/src/main/java/org/skriptlang/skript/bukkit/entity/player/elements/effects/EffBan.java
+++ b/src/main/java/org/skriptlang/skript/bukkit/entity/player/elements/effects/EffBan.java
@@ -78,9 +78,6 @@ public class EffBan extends Effect {
 	@Override
 	protected void execute(Event event) {
 		Component reason = this.reason == null ? null : this.reason.getSingle(event);
-		if (reason == null) {
-    		reason = Component.empty(); // DO check for null, a null reason generates NPE downstream!
-		}
 		Timespan duration = this.expires == null ? null : this.expires.getSingle(event);
 		Date expires = duration == null ? null : new Date(System.currentTimeMillis() + duration.getAs(TimePeriod.MILLISECOND));
 		for (Object object : players.getArray(event)) {
@@ -94,16 +91,14 @@ public class EffBan extends Effect {
 						String ip = address.getAddress().getHostAddress();
 						var banList = Bukkit.getBanList(Type.IP);
 						if (ban) {
-							String legacyReason = TextComponentParser.instance().toLegacyString(reason);
-							banList.addBan(ip, legacyReason, expires, SKRIPT_BAN_SOURCE);
+							banList.addBan(ip, toLegacyString(reason), expires, SKRIPT_BAN_SOURCE);
 						} else {
 							banList.pardon(ip);
 						}
 					} else {
 						var banList = Bukkit.getBanList(Type.NAME);
 						if (ban) {
-							String legacyReason = TextComponentParser.instance().toLegacyString(reason);
-							banList.addBan(player.getName(), legacyReason, expires, SKRIPT_BAN_SOURCE); // FIXME [UUID] ban UUID
+							banList.addBan(player.getName(), toLegacyString(reason), expires, SKRIPT_BAN_SOURCE); // FIXME [UUID] ban UUID
 						} else {
 							banList.pardon(player.getName());
 						}
@@ -119,8 +114,7 @@ public class EffBan extends Effect {
 					}
 					var banList = Bukkit.getBanList(Type.NAME);
 					if (ban) {
-						String legacyReason = TextComponentParser.instance().toLegacyString(reason);
-						banList.addBan(name, legacyReason, expires, SKRIPT_BAN_SOURCE);
+						banList.addBan(name, toLegacyString(reason), expires, SKRIPT_BAN_SOURCE);
 					} else {
 						banList.pardon(name);
 					}
@@ -129,7 +123,7 @@ public class EffBan extends Effect {
 					var ipBanList = Bukkit.getBanList(Type.IP);
 					var nameBanList = Bukkit.getBanList(Type.NAME);
 					if (ban) {
-						String legacyReason = TextComponentParser.instance().toLegacyString(reason);
+						String legacyReason = toLegacyString(reason);
 						ipBanList.addBan(ip, legacyReason, expires, SKRIPT_BAN_SOURCE);
 						nameBanList.addBan(ip, legacyReason, expires, SKRIPT_BAN_SOURCE);
 					} else {
@@ -152,6 +146,13 @@ public class EffBan extends Effect {
 			.appendIf(reason != null, "on account of", reason)
 			.appendIf(expires != null, "for", expires)
 			.toString();
+	}
+
+	private static @Nullable String toLegacyString(@Nullable Component component) {
+		if (component == null) {
+			return null;
+		}
+		return TextComponentParser.instance().toLegacyString(component);
 	}
 
 }

--- a/src/main/java/org/skriptlang/skript/bukkit/entity/player/elements/effects/EffBan.java
+++ b/src/main/java/org/skriptlang/skript/bukkit/entity/player/elements/effects/EffBan.java
@@ -77,7 +77,10 @@ public class EffBan extends Effect {
 
 	@Override
 	protected void execute(Event event) {
-		Component reason = this.reason == null ? null : this.reason.getSingle(event); // don't check for null, just ignore an invalid reason
+		Component reason = this.reason == null ? null : this.reason.getSingle(event);
+		if (reason == null) {
+    		reason = Component.empty(); // DO check for null, a null reason generates NPE downstream!
+		}
 		Timespan duration = this.expires == null ? null : this.expires.getSingle(event);
 		Date expires = duration == null ? null : new Date(System.currentTimeMillis() + duration.getAs(TimePeriod.MILLISECOND));
 		for (Object object : players.getArray(event)) {

